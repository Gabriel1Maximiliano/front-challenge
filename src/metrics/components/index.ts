export * from './CardInfoItem';
export * from './ChartLine';
export * from './PieChart';
export * from './SummaryMetrics';
