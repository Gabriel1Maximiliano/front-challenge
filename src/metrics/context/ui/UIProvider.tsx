/* eslint-disable @typescript-eslint/no-unsafe-assignment */
/* eslint-disable @typescript-eslint/no-explicit-any */
import { useReducer } from "react";
import { UiContext, uiReducer } from "./";

import { lightTheme } from "../../themes";

export interface UIState {
  isSideMenuOpen: boolean;
  currentMode: any;
}

const UI_Initial_State: UIState = {
  isSideMenuOpen: false,
  currentMode: lightTheme,
};

interface IProps {
  children: JSX.Element | JSX.Element[];
}

export const UIProvider = ({ children }: IProps): any => {
  const [state, dispatch] = useReducer(uiReducer, UI_Initial_State);

  const openSideMenu = (): void => {
    dispatch({ type: "[UI]-Open SideBar" });
  };
  const closeSideMenu = () => {
    dispatch({ type: "[UI]-Close SideBar" });
  };

  return (
    <UiContext.Provider
      value={{
        ...state,

        //Methods
        openSideMenu,
        closeSideMenu,
      }}
    >
      {children}
    </UiContext.Provider>
  );
};
