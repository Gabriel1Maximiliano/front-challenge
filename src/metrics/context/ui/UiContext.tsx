/* eslint-disable @typescript-eslint/no-explicit-any */

import { createContext } from "react";

export interface ContextProps {
  isSideMenuOpen: boolean;

  //Methods
  openSideMenu: () => void;
  closeSideMenu: () => void;
  
}

export const UiContext = createContext({} as ContextProps);
