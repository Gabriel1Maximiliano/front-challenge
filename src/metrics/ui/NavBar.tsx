import { UiContext } from "../context/ui";
import { useContext } from "react";

import { AppBar, IconButton, Toolbar, Typography } from "@mui/material";
import MenuIcon from "@mui/icons-material/Menu";
import DarkModeIcon from "@mui/icons-material/DarkMode";

export const NavBar = () => {
  const { openSideMenu } = useContext(UiContext);

  return (
    <AppBar
      position="sticky"
      elevation={0}
      sx={{ justifyContent: "space-between" }}
    >
      <Toolbar>
        <div style={{ display: "flex", alignItems: "center" }}>
          <IconButton size="large" onClick={openSideMenu}>
            <MenuIcon />
          </IconButton>
          <Typography style={{ marginLeft: "10px" }}>Challenge Meli</Typography>
        </div>
        <div style={{ marginLeft: "auto" }}>
          <IconButton >
            <DarkModeIcon />
          </IconButton>
        </div>
      </Toolbar>
    </AppBar>
  );
};
