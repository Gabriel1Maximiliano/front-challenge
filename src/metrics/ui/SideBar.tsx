/* eslint-disable @typescript-eslint/no-unsafe-member-access */
import {
  Box,
  Divider,
  Drawer,
  Typography,
} from "@mui/material";

import HttpIcon from "@mui/icons-material/Http";
import AutoGraphOutlinedIcon from "@mui/icons-material/AutoGraphOutlined";
import { UiContext } from "../context/ui";
import { useContext } from "react";


import DonutSmallOutlinedIcon from "@mui/icons-material/DonutSmallOutlined";
import QueryStatsOutlinedIcon from "@mui/icons-material/QueryStatsOutlined";
import FeedOutlinedIcon from "@mui/icons-material/FeedOutlined";
import { CardInfoItem } from "../components/CardInfoItem";

export const SideBar = () => {

  const { isSideMenuOpen, closeSideMenu } = useContext(UiContext);

  return (
    <Drawer anchor="left" open={isSideMenuOpen} onClose={closeSideMenu}>
      <Box sx={{ width: 250, ml: 3 }}>
        <Box sx={{ padding: "5px 10px" }}>
          <Typography>Menú</Typography>
        </Box>

        <Divider />
        <CardInfoItem
          urlToNavigate="/crear-url"
          icon={<HttpIcon />}
          itemText="Crear Url"
        />
        <CardInfoItem
          urlToNavigate="/historial"
          icon={<FeedOutlinedIcon />}
          itemText="Historial"
        />
        <CardInfoItem
          urlToNavigate="/estadisticas"
          icon={<AutoGraphOutlinedIcon />}
          itemText="Estadísticas"
        />
        <CardInfoItem
          urlToNavigate="/pie"
          icon={<DonutSmallOutlinedIcon />}
          itemText="Pie Chart"
        />
        <CardInfoItem
          urlToNavigate="/line"
          icon={<QueryStatsOutlinedIcon />}
          itemText="Line Chart"
        />

        <Divider />
      </Box>
    </Drawer>
  );
};
