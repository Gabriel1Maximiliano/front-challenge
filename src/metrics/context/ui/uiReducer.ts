import { UIState } from "./UIProvider";

type UIAtionType =
  | { type: "[UI]-Open SideBar" }
  | { type: "[UI]-Close SideBar" };

export const uiReducer = (state: UIState, action: UIAtionType): UIState => {
  switch (action.type) {
    case "[UI]-Close SideBar":
      return {
        ...state,
        isSideMenuOpen: false,
      };
    case "[UI]-Open SideBar":
      return {
        ...state,
        isSideMenuOpen: true,
      };

    default:
      return state;
  }
};
