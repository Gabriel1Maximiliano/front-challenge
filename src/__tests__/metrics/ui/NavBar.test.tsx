

/* eslint-disable @typescript-eslint/no-empty-function */
import {render} from '@testing-library/react'
import { NavBar } from '../../../metrics/ui'
describe('Pruebas sobre <NavBar />', () => {

    test('Debe hacer match con el SnapShot', ()=>{
        const { container } = render(<NavBar />);

        expect( container ).toMatchSnapshot();
    })
    test('Debe mostrar Challenge Meli en una etiqueta <p></p> ', ()=>{

        const title = 'Challenge Meli';
        const { getByText } = render(<NavBar />);

        expect( getByText(title) ).toBeTruthy();
    })
})

