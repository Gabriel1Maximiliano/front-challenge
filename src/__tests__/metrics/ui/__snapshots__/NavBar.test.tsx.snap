// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <NavBar /> Debe hacer match con el SnapShot 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionSticky css-12wl2ml-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    >
      <div
        style="display: flex; align-items: center;"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="MenuIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          style="margin-left: 10px;"
        >
          Challenge Meli
        </p>
      </div>
      <div
        style="margin-left: auto;"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="DarkModeIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9 9-4.03 9-9c0-.46-.04-.92-.1-1.36-.98 1.37-2.58 2.26-4.4 2.26-2.98 0-5.4-2.42-5.4-5.4 0-1.81.89-3.42 2.26-4.4-.44-.06-.9-.1-1.36-.1z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </header>
</div>
`;

exports[`Pruebas sobre <NavBar /> Debe mostrar Challenge Meli en una etiqueta <p></p>  1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionSticky css-12wl2ml-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    >
      <div
        style="display: flex; align-items: center;"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="MenuIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          style="margin-left: 10px;"
        >
          Challenge Meli
        </p>
      </div>
      <div
        style="margin-left: auto;"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="DarkModeIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9 9-4.03 9-9c0-.46-.04-.92-.1-1.36-.98 1.37-2.58 2.26-4.4 2.26-2.98 0-5.4-2.42-5.4-5.4 0-1.81.89-3.42 2.26-4.4-.44-.06-.9-.1-1.36-.1z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </header>
</div>
`;
